mp/src/game/shared/util_shared.cpp

```diff
@@ -296,7 +301,43 @@ bool CTraceFilterSimple::ShouldHitEntity( IHandleEntity *pHandleEntity, int cont
 	}
 
 	// Don't test if the game code tells us we should ignore this collision...
+#ifdef NEO
+	const CBaseEntity* pEntity = EntityFromEntityHandle(pHandleEntity);
+#else 
 	CBaseEntity *pEntity = EntityFromEntityHandle( pHandleEntity );
+#endif
+#ifdef NEO
+	if (m_collisionGroup == COLLISION_GROUP_PLAYER_MOVEMENT && m_pPassEnt && pEntity->IsPlayer())
+	{
+		const CBaseEntity *pEntity2 = EntityFromEntityHandle(m_pPassEnt);
+		if (pEntity2->IsPlayer())
+		{
+			if (sv_neo_collision_enemy_collisions.GetBool())
+			{
+				if (pEntity->GetTeamNumber() != pEntity2->GetTeamNumber())
+				{
+					return true;
+				}
+			}
+			if (sv_neo_collision_friendly_head_collisions.GetBool())
+			{
+				if (pEntity->GetAbsOrigin().z > pEntity2->GetAbsOrigin().z)
+				{ // pEntity is always the entity with a lower z value to pEntity 2
+					V_swap(pEntity, pEntity2);
+				}
+
+				if (pEntity2->GetAbsVelocity().z <= pEntity->GetAbsVelocity().z)
+				{ // If the entity on top is moving upwards faster than the entity on the bottom, ignore collision. Makes it easier for recons to bunnyhop out of spawn
+					float heightDifference = (pEntity2->GetAbsOrigin().z - (pEntity->GetAbsOrigin().z + pEntity->CollisionProp()->OBBMaxs().z));
+					if (heightDifference > sv_neo_collision_friendly_head_collisions_overlap.GetFloat())
+					{
+						return true;
+					}
+				}
+			}
+		}
+	}
+#endif
 	if ( !pEntity )
 		return false;
```

Comment:
Actually, are there scenarios pEntity and pEntity2 returns nullptr? The SDK code handles that scenario below.

Reply:
I don't know if there is any scenario where m_pPassEnt is true but pEntity2 isn't, but checking just in case